src/components/companies/Companies.tsx

```diff
@@ -1,59 +1,35 @@
 import { useEffect, useState } from "react"
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { Company } from "../../Interfaces";
+import { useUserLoggedContext } from "../../context/UserLoggedContext";
+import { fetchCompanies } from "../../trackerApiCalls";
 
-interface CompanyAttributes {
-  id: number;
-  name: string;
-  website: string;
-  street_address: string;
-  city: string;
-  state: string;
-  zip_code: string;
-  notes: string;
-}
-
-interface Company {
-  id: number;
-  type: string;
-  attributes: CompanyAttributes;
-}
 
 function Companies() {
   const [companies, setCompanies] = useState<Company[] | null>([]); 
   const [searchTerm, setSearchTerm] = useState("");
   const [filteredCompanies, setFilteredCompanies] = useState<Company[]>([]);
   const [isLoading, setIsLoading] = useState(true);
+  const navigate = useNavigate();
+  const {token, userData} = useUserLoggedContext();
 
+  
   useEffect(() => {
-    const fetchCompanies = async () => {
+    const getCompanies = async () => {
       try {
-        const token =
-          "";
-        const response = await fetch("http://localhost:3001/api/v1/users/2/companies", {
-          method: "GET",
-          headers: {
-            authorization: `Bearer ${token}`,
-            "Content-Type": "application/json"
-          },
-        });
-        console.log(response)
-
-        if (!response.ok) {
-          throw new Error(`Failed to fetch companies: ${response.statusText}`);
-        }
-
-        const data = await response.json();
-        console.log(data.data);
-        setCompanies(data.data as Company[]);
-        setFilteredCompanies(data.data as Company[]);
+        const companies = await fetchCompanies(userData.user.data.id, token!);
+        console.log("fetched companies", companies)
+        setCompanies(companies);
+        setFilteredCompanies(companies);
```

Comment:
Do you have anything that needs the json version of the response? If so, don't forget to convert it!